plugin/plugins/filesystem/nodes/utils.go

```diff
@@ -0,0 +1,197 @@
+package fsnodes
+
+import (
+	"context"
+	"hash/fnv"
+	"time"
+
+	"github.com/djdv/p9/p9"
+	"github.com/ipfs/go-cid"
+	ipld "github.com/ipfs/go-ipld-format"
+	logging "github.com/ipfs/go-log"
+	"github.com/ipfs/go-unixfs"
+	unixpb "github.com/ipfs/go-unixfs/pb"
+	coreiface "github.com/ipfs/interface-go-ipfs-core"
+	corepath "github.com/ipfs/interface-go-ipfs-core/path"
+)
+
+func doClone(names []string) bool {
```

Comment:
Hm. So, it looks like the example "localfs" filesystem implements this incorrectly.

---

Where does the 9p spec define "clone"? On linux, at least, clone is usually concerned with file descriptors. I believe it's sufficient to translate an empty path to ".".

Reply:
>So, it looks like the example "localfs" filesystem implements this incorrectly.

My interpretations may also be incorrect. I'll have to check again 😅
***

>Where does the 9p spec define "clone"? 

From 9P2000.L docs
>fid can be cloned to newfid by calling walk with nwname set to zero.

https://github.com/chaos/diod/blob/master/protocol.md

From 9P manual page
>The name “..” (dot-dot) represents the parent directory. The name “.” (dot), meaning the current directory, is not used in the protocol.
It is legal for nwname to be zero, in which case newfid will represent the same file as fid and the walk will usually succeed; this is equivalent to walking to dot. The rest of this discussion assumes nwname is greater than zero. 

https://9fans.github.io/plan9port/man/man9/walk.html

Reply:
Got it. So, really, we could just replace an empty "names" with a single dot. I'd consider doing that instead to make this case less special. Otherwise, we should add a comment explaining that (a) "" is equivalent to "." and (b) everything else in doClone/shouldClone is a performance optimization.

We also need to handle `././././.`. I don't think `PinFS` or the root filesystem is handling this correctly.